packages/edit-site/src/store/actions.js

```diff
@@ -125,6 +126,9 @@ export function setHomeTemplateId( homeTemplateId ) {
  * @return {number} The resolved template ID for the page route.
  */
 export function* setPage( page ) {
+	if ( ! page.path && page.context?.postId ) {
```

Comment:
Why do we need this?

Reply:
This `setPage` action tries to find the template which would be used to render the given page via the page path property like this: `example.com/path/to/page?_wp_find_template`. This basically relies on having a path to go to, and then once the template is resolved for that path, the request returns some JSON for the template instead of an HTML page like it normally would.

When we set pages via query args (`?postType=page&postId=2`), we don't have a path set. This means that the `setPage` action wouldn't receive a page path, and therefore would not be able to resolve the template which would be used to render that page.

So to get around that, if the path does not exist, but a post ID does exist, then we use the `?p=2` shorthand as the path so that it can resolve the template.